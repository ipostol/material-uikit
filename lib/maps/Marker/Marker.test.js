'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _enzyme = require('enzyme');

var _Marker = require('./Marker');

var _Marker2 = _interopRequireDefault(_Marker);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

describe('Marker component', function () {

  it('should render correctly', function () {

    var props = {
      isVisited: true
    };

    var wrapper = (0, _enzyme.shallow)(_react2.default.createElement(_Marker2.default, props));

    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly timeslotsCount=2, $hover=true', function () {

    var props = {
      timeslotsCount: 2,
      $hover: true
    };

    var wrapper = (0, _enzyme.shallow)(_react2.default.createElement(_Marker2.default, props));

    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly isLoosing=true', function () {

    var props = {
      isLoosing: true
    };

    var wrapper = (0, _enzyme.shallow)(_react2.default.createElement(_Marker2.default, props));

    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly isWinning=true', function () {

    var props = {
      isWinning: true
    };

    var wrapper = (0, _enzyme.shallow)(_react2.default.createElement(_Marker2.default, props));

    expect(wrapper).toMatchSnapshot();
  });
});