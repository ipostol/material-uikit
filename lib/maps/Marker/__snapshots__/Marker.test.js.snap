// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marker component should render correctly 1`] = `
<Tooltip
  align={
    Object {
      "points": Array [
        "tl",
        "br",
      ],
    }
  }
  arrowContent={null}
  destroyTooltipOnHide={false}
  mouseEnterDelay={0}
  mouseLeaveDelay={0.1}
  overlay={<Info />}
  placement="bottom"
  prefixCls="rc-tooltip-custom"
  trigger={
    Array [
      "hover",
    ]
  }
>
  <div
    className="container"
    style={null}
  >
    <div
      className="general visited"
    >
      <img
        alt=""
        className="icon"
      />
      £
      0
    </div>
    <div
      className="arrow visitedArrow"
    />
  </div>
</Tooltip>
`;

exports[`Marker component should render correctly isLoosing=true 1`] = `
<Tooltip
  align={
    Object {
      "points": Array [
        "tl",
        "br",
      ],
    }
  }
  arrowContent={null}
  destroyTooltipOnHide={false}
  mouseEnterDelay={0}
  mouseLeaveDelay={0.1}
  overlay={<Info />}
  placement="bottom"
  prefixCls="rc-tooltip-custom"
  trigger={
    Array [
      "hover",
    ]
  }
>
  <div
    className="container"
    style={null}
  >
    <div
      className="general loosing"
    >
      <img
        alt=""
        className="icon"
      />
      £
      0
    </div>
    <div
      className="arrow loosingArrow"
    />
  </div>
</Tooltip>
`;

exports[`Marker component should render correctly isWinning=true 1`] = `
<Tooltip
  align={
    Object {
      "points": Array [
        "tl",
        "br",
      ],
    }
  }
  arrowContent={null}
  destroyTooltipOnHide={false}
  mouseEnterDelay={0}
  mouseLeaveDelay={0.1}
  overlay={<Info />}
  placement="bottom"
  prefixCls="rc-tooltip-custom"
  trigger={
    Array [
      "hover",
    ]
  }
>
  <div
    className="container"
    style={null}
  >
    <div
      className="general winning"
    >
      <img
        alt=""
        className="icon"
      />
      £
      0
    </div>
    <div
      className="arrow winningArrow"
    />
  </div>
</Tooltip>
`;

exports[`Marker component should render correctly timeslotsCount=2, $hover=true 1`] = `
<Tooltip
  align={
    Object {
      "points": Array [
        "tl",
        "br",
      ],
    }
  }
  arrowContent={null}
  destroyTooltipOnHide={false}
  mouseEnterDelay={0}
  mouseLeaveDelay={0.1}
  overlay={
    <Info
      timeslotsCount={2}
    />
  }
  placement="bottom"
  prefixCls="rc-tooltip-custom"
  trigger={
    Array [
      "hover",
    ]
  }
>
  <div
    className="container"
    style={
      Object {
        "zIndex": 2,
      }
    }
  >
    <div
      className="general hover"
    >
      <img
        alt=""
        className="icon"
      />
      £
      0
      +
    </div>
    <div
      className="arrow hoverArrow"
    />
  </div>
</Tooltip>
`;
